
diff --git a/node_modules/cloudinary/lib/utils/index.js b/node_modules/cloudinary/lib/utils/index.js
index 45a49c4..e5f03d2 100644
--- a/node_modules/cloudinary/lib/utils/index.js
+++ b/node_modules/cloudinary/lib/utils/index.js
@@ -1071,7 +1071,11 @@
     let shasum = crypto.createHash('sha1');
     shasum.update(querystring.unescape(string_to_sign), 'utf8');
     return shasum.digest('hex');
-  }
+  },
+  api_sign_request: function(params_to_sign, api_secret) {
+    const to_sign = [];
+    for (let key in params_to_sign) {
+      if (params_to_sign.hasOwnProperty(key)) {
+        let value = params_to_sign[key];
+        if (Array.isArray(value)) {
+          value = value.join(',');
+        }
+        if (SHARED_JS_PARAMS.includes(key) || (value != null && value !== '')) {
+          to_sign.push(key + '=' + value);
+        }
+      }
+    }
+    const query_string = to_sign.sort().join('&');
+    return this.sha1(query_string + api_secret);
+  },
 };
 
 module.exports = utils;

